.storybook/stories/DataFiltering/components/DashboardPagination.tsx

```diff
@@ -6,26 +7,81 @@ import { useDataContext, useSortAndFilterContext } from '../lib/contexts';
 
 export const DashboardPagination = () => {
 	const { pagination, setPagination } = useSortAndFilterContext();
-	const { data, totalItems, totalPages } = useDataContext();
+	const { data, totalItems, totalPages, loading } = useDataContext();
+
+	const itemRangeText = generatePaginationRangeText({
+		totalItems: totalItems,
+		currentPage: pagination.page,
+		itemsPerPage: pagination.perPage,
+		singularNoun: 'audit',
+		pluralNoun: 'audits',
+	});
+
+	// Display text for pagination allows us to wait for loading to complete before updating
+	const [displayText, setDisplayText] = useState({
+		page: pagination.page,
+		itemRangeText,
+		perPage: pagination.perPage,
+	});
+
+	// Wait for loading to complete before updating pagination displayText
+	useEffect(() => {
+		if (loading) return;
+
+		setDisplayText({
+			page: pagination.page,
+			itemRangeText,
+			perPage: pagination.perPage,
+		});
+		// We only want to update update the display text once the table data has finished loading, all other deps are ignored here
+	}, [loading]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I wonder if we should still list all the other things as dependencies? How much extra work will it be doing since we're still returning early if it's loading

Reply:
I just explored this but it will still trigger the premature re-render of the component because those other values are updated before `loading` is. If we wanted to look at preventing abuse of the `useEffect` here, I could create an `onLoad` function which we pass to `useFetchData` and run that after `loading` is complete?

Reply:
I think it's fine